dask/base.py

```diff
@@ -1605,4 +1608,51 @@ def clone_key(key, seed):
     raise TypeError(f"Expected str or tuple[str, Hashable, ...]; got {key}")
 
 
+def print_on_compute(msg: str, val: T) -> T:
+    """When the value is computed, print it using distributed.print.
```

Comment:
This is confusing. What happens if you are running on the threads or multiprocessing scheduler? What happens if distributed is not installed?

Reply:
It will use regular `print`.